### PR TITLE
Allows any genitals on any gender

### PR DESCRIPTION
**Pull request by Multigamer99**
## About The Pull Request

Changelog
- Allows all genital options available for a given race to be available for any gender of that race.
    - Genitals are reset to the gender's default upon changing gender in the character creator.
- Cleans up sex code a bit (look i can only fix it so much rn)
    - Makes it dependent on genitals instead of gender in the case of humans. 
    - Vagina for female options.
    - Penis for male sex options.
    - Separates sucking and eating out to accommodate hermaphrodites
    - Separates jerking off and rubbing to accommodate hermaphrodites.
    - Horny mechanics have been partially unified and made dependent on genitals.
        - Anyone with balls can get blue balls now.
    - Getting someone pregnant requires you to have balls and them to have a vagina too, and given what it results from, also requires you to have a dick by proxy.
- Fixes dresses and some cloak slot items to cover genitals, such as tabards.

## Why It's Good For The Game

This cleans up some of the code a bit (not too much tbh) and fixes an oversight alongside adding a feature that many folks, including myself, have requested. This is a fantasy game, and so I don't think it's too crazy to allow genital customization. It's also nice for helping queer folks get more into the game, which is a plus too, and helps pave the way for things like gelding down the line.

I've also bug-tested this as much as my exhausted brain can do on my own, which is a plus (and trust me, with how much of a mess the sex code is, it needed bug testing).

### CODE REVIEW

File: code/modules/client/preferences_customizers.dm
Comment (Multigamer99) [reply code/modules/client/preferences_customizers.dm:225]: I tested it and it is. The customizer entries are not actually fully redone. I only added it because I noticed that changing from one to the other didn't update the parts.

Comment (Multigamer99) [reply code/modules/client/preferences_customizers.dm:225]: i'll touch it up if this pr turns out to not get denied.

File: code/modules/client/customizer/customizer.dm
Comment (Multigamer99) [reply code/modules/client/customizer/customizer.dm:35]: should test it but it's worth a shot.

note that the issue wasn't just "they stay enabled" but rather that it doesn't update them at all. staring on male gave a penis and testicles and switching to female still had just a penis and testicles, no vagina or chest were added, before my second proc was made.

**Comment by Multigamer99** [comment]
I think on the notes of "it doesn't fit the setting", I'd like to note that there is a huge amount of mixing and matching of body parts already in the setting with the wildkin, some animals have strange genital configurations or natural gender fluidity baked into them so some species make more sense with stuff beyond the basic male/female options, there's precedent for gender fuckery in other fantasy games like, for example, some versions of D&D having genderfluid kobold, and (according to discord discussion) surgery exists in the setting and thus could easily be how this is all done (and organ rejection isn't really a concern either plus donor parts aren't hard to come by in a place where lives are so cheap.).

Just wanted to note that here since it's a good extension on my reasons why it works for the game and setting since a final verdict has yet to be reached afaik.

**Comment by K4rlox** [comment]
>it doesnt fit the setting
doesnt fit the setting on a furry server mmm?

**Comment by Multigamer99** [comment]
I didn't strictly want to drag us into that mud pit but yeah, it's a furry server. One of the arguments on discord is that this was entirely just "fetish content" yet, at the same time, you could argue the inclusion of furries themselves is "fetish content" since, to many folks, they perceive only the parts of playing a furry character that relate to sex as having any substance when in reality it is a lot more than that. And yes, genitalia is a lot closer to a sexually focused topic, but it isn't exclusively one.

Plus, there is storytelling options with this. We already have kings who ban things like being gay, being straight, things based on bodily sex, marital status during sex, and so on. I 100% expect someone to make laws and the like regarding this at some point, and it would be interesting how folks handle laws regarding something that can be so easily hidden casually but can be tested for no matter what. It might also cause a potentially interesting hiccups in the cases where, for example, you have mercenaries disguised as tax men going around and taking payments in the form of coin or sex. Beyond that you have the ways this affects relationships, how it can subtly alter the mindset and thus roleplay, the way such things would alter hormone levels and thus could affect behaviour of the character in all manners of situations, and so on.

I also want to point out that this is a community that is horny as hell in ooc chat. Even if this was just "fetish content", that doesn't mean it isn't fitting, especially among other content additions which could be considered as such.

**Comment by Rin053** [comment]
One problem I can perceive with this immediately is that the queen can end up having a penis and perhaps the king doesn't want that. Or people having no way of discerning the gender of an individual on its face which may lead to some conversations being ended by '... by the way, do you have a penis?'

A pragmatic solution to this would be to make fates strictly determined by biological sex (genitals) as to what roles can be played / mixed and matched. So, for example, dickgirls would be treated as male. And if there isn't any visual indication, an approximate way of differentiating them through text. I'm not saying XE/XIR here, but perhaps something most people can agree on makes sense that you can see when you examine them so as to not confuse people.

**Comment by Multigamer99** [comment]
Making roles dependent on genitals is probably feasible. I'd have to go check some things but I would definitely be down for that if that's what the dev team wants. For the visibility thing, I'd rather not make it easily visible but I'm not against it either.

As for the the changes being moved to a separate PR, thanks! Glad to see it in the game so at least my bits aren't dangling out when I wear rags anymore.

**Comment by Multigamer99** [comment]
Also, I love going back and looking at the slow changes of thumbs ups and thumbs downs that have been occuring. Hoping this can be reviewed by Useroth before I gotta fix it due to a merge conflict from an update somewhere, and at least know if I should keep holding out hope or not. Still, glad there has been time to let more folks react.

**Comment by DeerJesus** [comment]
See the problem with this is I’ve never met a futanari RPer who wasn’t just a fucking MASSIVE gooner

**Comment by Multigamer99** [comment]
Isn't like half the playerbase massive gooners as it is?

Jokes aside, there's probably a correlation between players who play girls with bits and players who are horny but I don't think it's actually all that big of a deal, and said players are often capable of decent rp in my experience.

**Comment by FarsightedNightlight** [comment]
> "Server Vision"

As people have said already, this is a furry server. I won't make that my entire point of argument however, just because a woman has a dick doesn't make it any less dark fantasy. It's not like it magically exempts people from their rape-y fantasy content.

I personally have no issue with this, although this is also in mind with the fact my own characters tend to be gynomorph themselves. Saying it hurts server vision is just silly in line with people that say: "It has soul." It isn't a white and dark argument.

**Comment by Multigamer99** [comment]
That is a good point as well. Also add in that this is an inclusivity feature and, in my experience, folks that don't roleplay tend to be less accepting of servers that have features like that, which is good for us in this case.

Yeah, it is definitely a complicated debate. I am by no means unbiased myself: I want this in and I am 100% one of the horny players on the server. There's a reason I don't get to just merge this myself, and even if I could, I probably wouldn't and definitely shouldn't.

I think with the way the community is now? The everyone twirling gocks is not likely to happen IMO. Will a few folks do it? Probably, it's fucking SS13. That's basically inevitable. But I doubt it will significantly change the vibe of the server significantly beyond those few who like features like this because the mere presence of them makes them more comfortable and the ERPers.

That's just my opinion though, which is... Well it's an opinion.

**Comment by Gyrandola** [comment]
> ~I simultaneously don't like the PR but I have to admit if this got merged I'd be the first person to make a dick girl character. I sadly have to downvote to uphold server vision.~


God talked to me in my sleep. I had an epiphany and realized I'm actually neutral on this. We'll see what Useroth says.

**Comment by dessysalta** [comment]
> the queen can end up having a penis and perhaps the king doesn't want that

In fairness, the queen is the queen consort and isn't always actually in love with the king and vice versa. She's been married to for political value and to act as another advisor/heiress to the throne if things go bad.

Not always the case, but worth mentioning. 

> '... by the way, do you have a penis?'

If a character unironically says this to mine regardless of whether or not my character is trans/has mismatched genitals I'm going to laugh in their face so hard I fall over and instantly cross with the undermaiden.

> make fates strictly determined by biological sex (genitals)

Not a great idea, but not terrible? This means someone who looks, acts, and sounds male wouldn't be allowed to serve as a man at arms, and similar. I suggest there be a small blurb if you look at them with nothing more than pants covering their lower half, something along the lines of "their groin appears to betray their appearance," something light and generally inoffensive. Double points if it can be triggered by having something in your belt or whatever, because then you can unironically do the whole "is that a knife in your pants or are you just happy to- BY THE NINE!" bit.

**Comment by hivemuscle** [comment]
retarded fetish pr, slowly turning game into splurt 2.0 but middle age setting

**Comment by Rin053** [comment]
> If a character unironically says this to mine regardless of whether or not my character is trans/has mismatched genitals I'm going to laugh in their face so hard I fall over and instantly cross with the undermaiden.

I'm more neutral on this though certain dialogue may seem uncanny when someone doesn't know who they're talking to. All 'hiding the gender' does, really, is just make people make a mental note OOCly to engage or disengage with that particular conversation after knowing (if you play statics).

> Not a great idea, but not terrible? This means someone who looks, acts, and sounds male wouldn't be allowed to serve as a man at arms, and similar.

The pragmatic approach was that 'Head of Household' and similar was dependent on whether or not you were the bearer or the producer of heirs. You can't really be a King if you have a vagina and this role is more determined by your ability to produce heirs rather than your physical appearance. For combat roles like men at arms it would be more a contest of physical ability, which is why they should just be considered on the status of 'men'. There's perhaps an entire argument to be had about giving them their own roles between man and woman but I'm not going to figure that one out. This is just the most straightforward way I can see. In reality it requires people to accept this game isn't medieval times, it's a fantasy roleplaying game that borrows from these elements.

**Comment by Zebraman** [comment]
It's hilarious to watch ppl complain about this shit while someone made a PR to add a playable goblin race (literally evil little demons that shall be hunted down by everyone in the lore) and it got instant merged.

**Comment by KKLZ4** [comment]
Firmly against this.

This is just a slow, steady step from Roguetown into Medieval Skyrat/Nova/God-forbid SPLURT. ERPers might not bother anybody, but they also don't fucking /do/ anything, and alot of them seriously struggle to actually fulfill their role's requirements (Especially the ones who play guards and exclusively go around ERPing maids or whatever.) We'll inevitably confuse people "It's an ERP server because advanced sexcode (Which was never intended to be anything more than a meme to begin with) futas and furries" and then the RogueTown players going around actually playing the codebase like it was classic RT.

This is not an ERP server.

**Comment by optimumtact** [comment]
> This is not an ERP server.

Bro is so far in denial he's in egypt

**Comment by ChungusGamer666** [comment]
This PR is still bad and Oranges should be beheaded

**Comment by Trubbl3** [comment]
as much i would like this change i know it shouldn't be done

**Review by vitorhks — APPROVED** [review]

**Comment by Multigamer99** [comment]
oh hey someone approved of this

i'll fix the merge conflicts after the current round then

**Comment by Multigamer99** [comment]
> > oh hey someone approved of this
> > i'll fix the merge conflicts after the current round then
> 
> If it's not a green checkmark then it's not a person with permissions on the repo, anyone can approve like that

Oh, RIP

Got my hopes up for nothing.

**Comment by Elli-Skala** [comment]
All balls, no dick

**Comment by Multigamer99** [comment]
> All balls, no dick

Yeah no that's technically an option.

**Comment by dgen44** [comment]
Honestly this changes the dynamic of a server entirely. On every server where futanari was allowed, it tends to make up a majority of the population. When given the option to fuck the big-breasted brawny she-stud or the average masculine character, most people are going to pick the futa and then everyone is gonna want to play futa, because everyone is going to gravitate towards that. 

Also a majority of futa players fail to properly represent a mixed-gender character and instead just hypersexualize the fuck out of their characters as overzealous and toxic people that feel the need to turn erp into a competitive scene. 

It would be nice if people could be trusted to use this PR properly, but it’s honestly gross how most futa players misrepresent and oversexualize trans/etc.

This is a fetish PR that does nothing to enrich roleplay climate.
Might as well add limitless breast/ass/dick sizes like splurt.

**Comment by Useroth** [comment]
Hey. It's probably time I give a definitive answer for it.
Won't be definitely won't be merging this in the form of "everyone can pick whatever".
It may come back later on in some form, but it will not be anything overly widespread. I'll currently close the PR.

**Comment by Multigamer99** [comment]
Thanks for giving that verdict. I'll look into options like making it buyable with triumphs or something for a potential comeback later.

**Comment by Quink-Fix** [comment]
Any update on a follow-up to this?

**Comment by Multigamer99** [comment]
That would be a thing to ask staff about directly, especially since I am not really involved with the server anymore and am unlikely to get unbanned at this rate

In hindsight this pr had some issues found in further testing although I ended up managing to iron those out when I ported it for hearthstone. Shouldn't be too hard to port it back to ratwood from there, and I have a vague idea for how to do it but unless a close friend asks me to or ratwood staff want it done (and maybe an incentive to do it for a server I can't play on) I likely won't be the one to do it. If someone else takes it on and wants advice to prevent some of the bugs/issues that came up, I will gladly help with that though.

> Any update on a follow-up to this?

So in short, no. There won't be a follow up from me in the forceable future, but this has been followed up on/implemented on other codebases/servers, such as hearthstone and its downstreams.

 \- Vrell

**Pull request closed without merging**
